packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js

```diff
@@ -185,4 +185,54 @@ describe('TaskNavButtonsContainer', function () {
       expect(createDefaultAnnotationSpy.notCalled).to.be.true
     })
   })
+
+  describe('#onSubmit', function () {
+    let wrapper
+    let completeClassificationSpy
+    let createDefaultAnnotationIfThereIsNoneSpy
+    let onSubmitSpy
+    const preventDefaultSpy = sinon.spy()
+
+    before(function () {
+      completeClassificationSpy = sinon.spy()
+      createDefaultAnnotationIfThereIsNoneSpy = sinon.spy(TaskNavButtonsContainer.wrappedComponent.prototype, 'createDefaultAnnotationIfThereIsNone')
+      onSubmitSpy = sinon.spy(TaskNavButtonsContainer.wrappedComponent.prototype, 'onSubmit')
+
+      wrapper = shallow(
+        <TaskNavButtonsContainer.wrappedComponent
+          completeClassification={completeClassificationSpy}
+          isThereAPreviousStep={() => { }}
+          isThereANextStep={() => { }}
+          tasks={tasks}
+        />
+      )
+    })
+
+    afterEach(function () {
+      completeClassificationSpy.resetHistory()
+      createDefaultAnnotationIfThereIsNoneSpy.resetHistory()
+      onSubmitSpy.resetHistory()
+      preventDefaultSpy.resetHistory()
+    })
+
+    after(function () {
+      createDefaultAnnotationIfThereIsNoneSpy.restore()
+      onSubmitSpy.restore()
+    })
+
+    it('should prevent the event default', function () {
+      wrapper.instance().onSubmit({ preventDefault: preventDefaultSpy })
+      expect(preventDefaultSpy).to.have.been.calledOnce
+    })
+
+    it('should call createDefaultAnnotationIfThereIsNone', function () {
+      wrapper.instance().onSubmit({ preventDefault: preventDefaultSpy })
+      expect(createDefaultAnnotationIfThereIsNoneSpy).to.have.been.calledOnce
+    })
```

Comment:
I dislike testing code directly like this, since your test is tied directly to the implementation. I'd prefer to call `onSubmit` then check whether a mock classification has a valid annotations array, as set by a mock `createDefaultAnnotation` action.

I think that would also test whether your code overwrites any existing task annotations. I'm sure it doesn't, but this test doesn't check for that outcome.

Reply:
That behavior is being tested in the block for `createDefaultAnnotationIfThereIsNone`. I think it makes sense here to test that `createDefaultAnnotationIfThereIsNone` is called only once since these are the unit tests for the event handler.

Perhaps we should try to get on the same page about when we should do unit tests versus integration tests? If I add an expectation here for the outcome of `createDefaultAnnotationIfThereIsNone`, it feels like duplicating something already tested.